### DISCUSSION TITLE
@sanity/image-url fails to load environment variables

### DISCUSSION THREAD

**Discussion by Arctomachine** [sanity-io/sanity#3691] [Sanity Studio Support] [answered]
I am using Next as frontend. With `@sanity/image-url`, if `next-sanity` client config is loaded from environment variables, then it fails with following error:
```
Unhandled Runtime Error

Error: Configuration must contain `projectId`
```
If values of config are hardcoded, it runs fine. Without this package client also runs fine from environment variables. Is there a way to make it work?

Here is my related code:
```tsx
// index.tsx
import createSanityClient from '../lib/createSanityClient'
import imageUrlBuilder from '@sanity/image-url'

const sanityClient = createSanityClient()
const sanityImageBuilder = imageUrlBuilder(sanityClient)
```
```ts
// createSanityClient.ts
import { createClient } from 'next-sanity'

export default function createSanityClient () {
  return createClient({
    projectId: process.env.PROJECT_ID,
    dataset: process.env.DATASET,
    apiVersion: '2021-10-21',
    useCdn: false,
  })
}
```
Context info:
I declare client and builder as top level variables inside `index.tsx`. Then I use client inside `getStaticProps` to run few queries and pass results to main function. Then inside main function I cycle through array `services` from query result and return components with images like this `<Service ... background={ sanityImageBuilder.image(service.image).url() } ... />`.

**Comment by geball** [comment, marked as answer]
Hi @Arctomachine. Could you please try exposing the environment variables to the client using the `NEXT_PUBLIC_` prefix (as done [here](https://github.com/sanity-io/next-sanity/blob/main/sanity.config.ts)) to see if that helps?

**Reply by Arctomachine** [reply]
Yes, it worked
